### PR TITLE
Update to Rails 4.2

### PR DESCRIPTION
**Pull request by cypher**
Implements #28.

Opening this now to facilitate review and questions, don't merge this yet.

### CODE REVIEW

File: config/initializers/mailer.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Rails.application.config.action_mailer.delivery_method = :smtp
```
Comment (carpodaster) [inline config/initializers/mailer.rb]: Can you exclude the test env from this setting?

**Comment by carpodaster** [comment]
Awesome stuff, @cypher ! In only noticed that the update would start sending mails during test runs which is probably not what we want. Other than that, looks great! #yolomerge at will :)

**Comment by cypher** [comment]
@carpodaster D'oh, I knew there was a reason why this wasn't in an initializer in the first place -_-

Also, there's one more thing that should be done, which is move the `secret_token` out of `config/initializers/secret_token.rb`, and into an environment variable called `SECRET_KEY_BASE`. This needs to be done by whoever deploys it. Ideally, you would also set a completely new token, since the old one is available here in the repo, and (in theory) could lead to trouble down the road. And since we don't seem to store much stuff in the cookie anyway, making all existing ones invalid shouldn't be a problem.

I can do the (re)moving part, but I'm guessing you or @svenfuchs will have to set up the environment variable?

**Comment by carpodaster** [comment]
@cypher I've just set a `SECRET_KEY_BASE` ENV var with some renewed randomness

**Comment by cypher** [comment]
:+1: I've just removed the `secret_token.rb` file.

**Comment by carpodaster** [comment]
Hey @cypher – are we cool to merge this? This would be so great! :)

**Comment by cypher** [comment]
Yeah, I think this is good to merge. Go ahead!

**Comment by alicetragedy** [comment]
just on time for [Rails 5.0.0.rc1](http://weblog.rubyonrails.org/2016/5/6/this-week-in-rails-railsconf-recap-rails-5-0-rc-1-is-out/)! :trollface:

**Comment by cypher** [comment]
Let's just wait for the final version ;)

**Comment by alicetragedy** [comment]
hey @carpodaster @cypher, I got an email from Travis saying that the build has errored. When I log into the heroku console, it says `Loading production environment (Rails 3.2.22)` so.. the merge was unsuccessful? I've never seen this happen before.
Have not looked at this in-depth yet but [this](https://travis-ci.org/rails-girls-summer-of-code/rgsoc-campaign/builds/130926760) seems to be the problem 🤔

**Comment by cypher** [comment]
@alicetragedy huh. Apparently, I forgot to check in the updated `Gemfile.lock`, but I don't remember it showing up in `git status` as changed.

I've pushed an updated version, and hopefully it will build now.

**Comment by alicetragedy** [comment]
@cypher it did! 💃 👯

**Comment by cypher** [comment]
😸

**Pull request merged**